src/main/java/com/zaxxer/hikari/util/PropertyElf.java

```diff
@@ -180,4 +186,29 @@ private static String capitalizedPropertyName(String propertyName)
       // use the english locale to avoid the infamous turkish locale bug
       return propertyName.substring(0, 1).toUpperCase(Locale.ENGLISH) + propertyName.substring(1);
    }
+
+   private static Optional<Duration> parseDuration(String value)
+   {
+      Matcher matcher = DURATION_PATTERN.matcher(value);
+      if (matcher.matches()) {
+         long number = Long.parseLong(matcher.group("number"));
+         String unit = matcher.group("unit");
+         switch (unit) {
+            case "ms":
+               return Optional.of(Duration.ofMillis(number));
+            case "s":
+               return Optional.of(Duration.ofSeconds(number));
+            case "m":
+               return Optional.of(Duration.ofMinutes(number));
+            case "h":
+               return Optional.of(Duration.ofHours(number));
+            case "d":
+               return Optional.of(Duration.ofDays(number));
+            default:
+               throw new IllegalStateException(String.format("Could not match unit, got %s (from given value %s)", unit, value));
```

Comment:
Shouldn't this rather be an IllegalArgumentException?

Reply:
I made it an `IllegalStateException` because the regular expression should already catch any non-matching value (the units are part of the regular expression). So if it ever hits the `default` clause, there is something wrong with the code, not the user's input value.

However I am entirely fine with making it `IllegalArgumentException` if that is preferred.